File: interview/01.01.go
/*
 * @Author: lmio
 * @Date: 2023-04-10 22:56:21
 * @LastEditTime: 2023-04-10 23:02:11
 * @FilePath: /leetcode/interview/01.01.go
 * @Description:面试题 01.01. 判定字符是否唯一
 */
package interview

func IsUnique(astr string) bool {
	code := make([]bool, 26)
	if len(astr) > 26 {
		return false
	}
	for i := range astr {
		if code[astr[i]-'a'] {
			return false
		}
		code[astr[i]-'a'] = true
	}
	return true
}

File: interview/01.01.py
'''
Author: lmio 2091319361@qq.com
Date: 2023-08-31 19:25:52
LastEditors: lmio 2091319361@qq.com
Description: 面试题 01.01. 判定字符是否唯一
'''

class Solution:
    def isUnique(self, astr: str) -> bool:
        vis = [False]*26
        for ch in astr:
            key = ord(ch) - ord('a')
            if vis[key]:
                return False
            vis[key] = True
        return True